src/send_kcidb.py

```diff
@@ -8,126 +8,109 @@
-                    created_time = datetime.datetime.fromtimestamp(
-                        created_time.timestamp(), tz=tz_utc)
-                revision = {
-                    "builds": [],
-                    "checkouts": [{
-                        "id": f"kernelci:{node['_id']}",
-                        "origin": "kernelci",
-                        "tree_name": node["revision"]["tree"],
-                        "git_repository_url": node["revision"]["url"],
-                        "git_commit_hash": node["revision"]["commit"],
-                        "git_repository_branch": node["revision"]["branch"],
-                        "start_time": created_time.isoformat(),
-                        "patchset_hash": "",
-                    }],
-                    "tests": [],
-                    "version": {
-                        "major": 4,
-                        "minor": 0
-                    }
-                }
-                self.send_revision(client, revision)
+class KCIDBBridge(Service):
+    def __init__(self, configs, args, name):
+        super().__init__(configs, args, name)
 
-            sys.stdout.flush()
+    def _setup(self, args):
+        return {
+            'client': Client(
+                project_id=args.kcidb_project_id,
+                topic_name=args.kcidb_topic_name
+            ),
+            'sub_id': self._db.subscribe_node_channel(filters={
+                'name': 'checkout',
+                'state': 'done',
+            })
+        }
 
-        except KeyboardInterrupt:
-            self._logger.log_message(logging.INFO, "Stopping.")
-        except Exception:
-            self._logger.log_message(logging.ERROR, traceback.format_exc())
-        finally:
-            db.unsubscribe(sub_id)
+    def _stop(self, context):
+        if context['sub_id']:
+            self._db.unsubscribe(context['sub_id'])
 
-        sys.stdout.flush()
+    def _send_revision(self, client, revision):
+        if kcidb.io.SCHEMA.is_valid(revision):
+            return client.submit(revision)
+        self.log.error("Aborting, invalid data")
 
+    @staticmethod
+    def _set_timezone(created_timestamp):
+        created_time = datetime.datetime.fromisoformat(created_timestamp)
+        if not created_time.tzinfo:
+            tz_utc = datetime.timezone(datetime.timedelta(hours=0))
+            created_time = datetime.datetime.fromtimestamp(
+                created_time.timestamp(), tz=tz_utc)
+        return created_time.isoformat()
+
+    def _run(self, context):
+        self.log.info("Listening for events... ")
+        self.log.info("Press Ctrl-C to stop.")
+
+        while True:
+            node = self._db.receive_node(context['sub_id'])
+            self.log.info(f"Submitting node to KCIDB: {node['_id']}")
+
+            revision = {
+                'builds': [],
+                'checkouts': [{
+                    'id': f"kernelci:{node['_id']}",
+                    'origin': 'kernelci-api',
```

Comment:
Note: I think it would be good to have a `--origin` command line option, as a follow-up.

Reply:
Let me do it in this PR itself as `origin` has been added in this one.

Reply:
The `origin` field was already there as `kernelci`, but yes this PR is changing it.  I guess we can leave `kernelci` by default but set `kernelci-api` in the staging configuration file in `kernelci-deploy`.

Reply:
Ah I see you've already added it.  It might have been better as a separate commit as that's more than just refactoring the code to use the Service base class.

Reply:
Yes, I'll add a separate commit.

Reply:
Also, corrected `checkouts.id` field to use `origin` in the value to fix invalid data error.